base/abstractarray.jl

```diff
@@ -2351,7 +2312,7 @@ function _typed_hvncat(::Type{T}, ::Val{N}, as::AbstractArray...) where {T, N}
         end
     end
 
-    A = cat_similar(as[1], T, (ntuple(d -> size(as[1], d), N - 1)..., Ndim, ntuple(x -> 1, nd - N)...))
+    A = cat_similar(as[1], T, (ntuple(d -> size(as[1], d), N - 1)..., Ndim, ntuple(x -> 1, nd - N)...), Val(nd))
```

Comment:
this (and one above it) seem to already be a tuple. If it isn't inferring that size, it might need a type assert on that tuple, not Val(nd) here

Reply:
Ah, I see what's going on. Looks like it was inferring fine here, so I'll strip that.